specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,295 @@
+                    m_appWindow->GetWebViewEnvironment()->CreateWebResourceResponse(
+                        nullptr, 404, L"Not Found", L"", &response));
+                CHECK_FAILURE(args->put_Response(response.get()));
+            }
+            return S_OK;
+        }
+
+        return S_OK;
+        })
+        .Get(),
+    &m_webResourceRequestedToken));
+
+CHECK_FAILURE(m_webView->Navigate(L"https://www.example.com"));
+
+// The following XHR will execute in the context of https://www.example.com page and will succeed.
+// WebResourceRequested event will be raised for this request as *.example.com is in the allowed origin list of custom-scheme. 
+// Since the response header provided in WebResourceRequested handler allows all origins for CORS
+// the XHR succeeds.
+CHECK_FAILURE(m_webView->ExecuteScript(L"var oReq = new XMLHttpRequest();"
+                                       L"oReq.addEventListener(\"load\", reqListener);"
+                                       L"oReq.open(\"GET\", \"custom-scheme://example-data.json\");"
+                                       L"oReq.send();",
+                                      Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                                          [](HRESULT error, PCWSTR result) -> HRESULT { return S_OK; })
+                                          .Get()));
+// The following XHR will fail because *.example.com is not in the allowed origin list of custom-scheme2.
+// The WebResourceRequested event will not be raised for this request.
+CHECK_FAILURE(m_webView->ExecuteScript(L"var oReq = new XMLHttpRequest();"
+                                       L"oReq.addEventListener(\"load\", reqListener);"
+                                       L"oReq.open(\"GET\", \"custom-scheme-not-in-allowed-origins://example-data.json\");"
+                                       L"oReq.send();",
+                                      Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                                          [](HRESULT error, PCWSTR result) -> HRESULT { return S_OK; })
+                                          .Get()));
+```
+
+# API Details
+
+## COM API
+
+```c#
+// This is the ICoreWebView2CustomSchemeRegistration interface
+[uuid(d60ac92c-37a6-4b26-a39e-95cfe59047bb), object, pointer_default(unique)]
+interface ICoreWebView2CustomSchemeRegistration : IUnknown {
+  // Represents the registration of a custom scheme with the CoreWebView2Environment.
+  // This allows the WebView2 app to be able to handle
+  // WebResourceRequested event for requests with the specified scheme and
+  // be able to navigate the WebView2 to the custom scheme. Once the environment
+  // is created, the registrations are valid and immutable throughout the
+  // lifetime of the associated WebView2s' browser process and any WebView2
+  // environments sharing the browser process must be created with identical
+  // custom scheme registrations, otherwise the environment creation will fail.
```

Comment:
See earlier question about what this sentence is actually saying.

Reply:
This is correct. See above.